### DISCUSSION TITLE
Need to get the list of vm guests using a specific datastore

### DISCUSSION THREAD

**Discussion by packetq** [vmware/govmomi#3133] [Community Help] [answered]
Hi,

I need to get the list of vm guests that are using a particular datastore. I'm supposed to get back a slice of references to the vm guests. I'm having some trouble trying to figure out what to do with those references (ie; access the vm name). I have been looking at the source, but I haven't found anything that sheds any light on this. Can someone please point me in the right direction?

Thanks
-G

**Comment by dougm** [comment, marked as answer]
Have you looked at using the property collector?  See example below, where `host.Vm` is a slice of references: https://github.com/vmware/govmomi/blob/de8a56d089b7c1b65cf14178c2c2a3fca80db914/property/example_test.go#L69-L70
